assembly/compliance-tool/src/test/java/org/eclipse/sw360/antenna/frontend/compliancetool/sw360/updater/SW360UpdaterTestWithFailure.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) Bosch.IO GmbH 2020.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.sw360.antenna.frontend.compliancetool.sw360.updater;
+
+
+import org.assertj.core.api.Assertions;
+import org.eclipse.sw360.antenna.frontend.compliancetool.sw360.ComplianceFeatureUtils;
+import org.eclipse.sw360.antenna.frontend.compliancetool.sw360.SW360Configuration;
+import org.eclipse.sw360.antenna.frontend.compliancetool.sw360.SW360TestUtils;
+import org.eclipse.sw360.antenna.model.artifact.facts.ArtifactClearingState.ClearingState;
+import org.eclipse.sw360.antenna.sw360.client.adapter.AttachmentUploadResult;
+import org.eclipse.sw360.antenna.sw360.client.adapter.SW360Connection;
+import org.eclipse.sw360.antenna.sw360.client.adapter.SW360ReleaseClientAdapter;
+import org.eclipse.sw360.antenna.sw360.client.rest.resource.releases.SW360Release;
+import org.eclipse.sw360.antenna.sw360.client.utils.SW360ClientException;
+import org.eclipse.sw360.antenna.sw360.workflow.generators.SW360UpdaterImpl;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.Objects;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * If during the generation of the clearing document an exception is thrown, the uspdater must
+ * carry on and propagate the error at the end of the process
+ * @see <a href="https://github.com/eclipse/antenna/issues/575">original bug</a>
+ */
+public class SW360UpdaterTestWithFailure {
```

Comment:
What is the reason again why the testcase in this class cannot be added to the existing _SW360UpdaterTest_ class? There is now duplicate initialization logic.

Reply:
The SW360UpdaterTest is slightly different because it works with and empty CSV file (which is wrong but we have a ticket for that https://github.com/eclipse/antenna/issues/593)

Reply:
So when the ticket is solved, the tests could be merged again?

Reply:
yes. Shall I put a todo in both classes ?

Reply:
Yes, this would be fine.

Reply:
done